lib/search/query.rb

```diff
@@ -81,5 +85,22 @@ def content_index_names
       # index is a IndexForSearch object, which combines all the content indexes
       index.index_names
     end
+
+    def fetch_spell_checks(search_params)
+      SpellCheckFetcher.new(search_params, registries).es_response
+    end
+
+    def process_elasticsearch_errors
+      yield
+    rescue Elasticsearch::Transport::Transport::Errors::BadRequest => e
+      case e.message
+      when /Numeric value \([0-9]*\) out of range of/
+        raise(NumberOutOfRange, 'Integer value exceeds maximum allowed')
```

Comment:
Maybe we could include the out-of-range number in the error message? It looks like we can extract that easily even if it would take a lot of effort to parse the field name.